File: mobile/lib/services/category_service.dart
import '../models/category.dart';
import 'api_client.dart';

class CategoryService {
  final ApiClient _client = ApiClient();

  Future<List<Category>> getAll() async {
    final data = await _client.get('/categories');
    return (data as List).map((e) => Category.fromJson(e)).toList();
  }

  Future<Category> create(String name) async {
    final data = await _client.post('/categories', {'name': name});
    return Category.fromJson(data);
  }

  Future<Category> update(String id, String name) async {
    final data = await _client.put('/categories/$id', {'name': name});
    return Category.fromJson(data);
  }

  Future<void> delete(String id) async {
    await _client.delete('/categories/$id');
  }
}

File: web-admin/src/app/core/services/category.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Category } from '../models/models';

@Injectable({ providedIn: 'root' })
export class CategoryService {
  private http = inject(HttpClient);
  private base = `${environment.apiBaseUrl}/categories`;

  getAll(): Observable<Category[]> {
    return this.http.get<Category[]>(this.base);
  }

  create(name: string): Observable<Category> {
    return this.http.post<Category>(this.base, { name });
  }

  update(id: string, name: string): Observable<Category> {
    return this.http.put<Category>(`${this.base}/${id}`, { name });
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }
}
